`ST_FILESYSTEM_AVAILABILITY` macro definitions breaks builds with gcc on macOS

@zrax This part breaks builds of dependents with gcc, and also seems just incorrect, since filesystem is provided by C++ runtime, not the OS: https://github.com/zrax/string_theory/blob/43bbb4fb14e6e9bb8dee7ca7abb7f2a4fdf1dddb/include/st_config.h.in#L56-L60

This is what happens:
```
[  0%] Building CXX object _deps/catch2-build/src/CMakeFiles/Catch2.dir/catch2/benchmark/detail/catch_run_for_at_least.cpp.o
cd /opt/local/var/macports/build/libhsplasma-b025405b/work/build/_deps/catch2-build/src && /opt/local/bin/g++-mp-14 -DHS_BIG_ENDIAN -DMACOSX -I/opt/local/var/macports/build/libhsplasma-b025405b/work/build/_deps/catch2-src/src/catch2/.. -I/opt/local/var/macports/build/libhsplasma-b025405b/work/build/_deps/catch2-build/generated-includes -Wall -Wextra -Wno-unused-parameter -pipe -I/opt/local/libexec/openssl3/include -Os -DNDEBUG -I/opt/local/libexec/openssl3/include -isystem/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++17 -arch ppc -mmacosx-version-min=10.6 -fvisibility=hidden -fvisibility-inlines-hidden -ffile-prefix-map=/opt/local/var/macports/build/libhsplasma-b025405b/work/build/_deps/catch2-src/= -MD -MT _deps/catch2-build/src/CMakeFiles/Catch2.dir/catch2/benchmark/detail/catch_run_for_at_least.cpp.o -MF CMakeFiles/Catch2.dir/catch2/benchmark/detail/catch_run_for_at_least.cpp.o.d -o CMakeFiles/Catch2.dir/catch2/benchmark/detail/catch_run_for_at_least.cpp.o -c /opt/local/var/macports/build/libhsplasma-b025405b/work/build/_deps/catch2-src/src/catch2/benchmark/detail/catch_run_for_at_least.cpp
In file included from /opt/local/include/string_theory/st_assert.h:24,
                 from /opt/local/include/string_theory/st_charbuffer.h:24,
                 from /opt/local/include/string_theory/st_string_priv.h:24,
                 from /opt/local/include/string_theory/st_string.h:27,
                 from /opt/local/include/string_theory/st_formatter.h:24,
                 from /opt/local/include/string_theory/st_format.h:24,
                 from /opt/local/include/string_theory/format:1,
                 from /opt/local/var/macports/build/libhsplasma-b025405b/work/libhsplasma-2025.11.04/core/Debug/plDebug.h:20,
                 from /opt/local/var/macports/build/libhsplasma-b025405b/work/libhsplasma-2025.11.04/core/Debug/plDebug.cpp:17:
/opt/local/include/string_theory/st_string.h:308:51: error: clause 'strict' is not recognised for the 'availability' attribute
  308 |         string(const std::filesystem::path &path) ST_FILESYSTEM_AVAILABILITY
      |                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/include/string_theory/st_string.h:503:53: error: clause 'strict' is not recognised for the 'availability' attribute
  503 |         void set(const std::filesystem::path &path) ST_FILESYSTEM_AVAILABILITY
      |                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/include/string_theory/st_string.h:688:62: error: clause 'strict' is not recognised for the 'availability' attribute
  688 |         string &operator=(const std::filesystem::path &path) ST_FILESYSTEM_AVAILABILITY
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/include/string_theory/st_string.h:946:68: error: clause 'strict' is not recognised for the 'availability' attribute
  946 |         static string from_path(const std::filesystem::path &path) ST_FILESYSTEM_AVAILABILITY
      |                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/include/string_theory/st_string.h:1193:47: error: clause 'strict' is not recognised for the 'availability' attribute
 1193 |         std::filesystem::path to_path() const ST_FILESYSTEM_AVAILABILITY
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/include/string_theory/st_formatter.h:621:64: error: clause 'strict' is not recognised for the 'availability' attribute
  621 |                             const std::filesystem::path &path) ST_FILESYSTEM_AVAILABILITY
      |                                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/local/include/string_theory/st_stringstream.h:283:70: error: clause 'strict' is not recognised for the 'availability' attribute
  283 |         string_stream &operator<<(const std::filesystem::path &path) ST_FILESYSTEM_AVAILABILITY
      |                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [core/CMakeFiles/HSPlasma.dir/Debug/plDebug.cpp.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/libhsplasma-b025405b/work/build'
make[1]: *** [core/CMakeFiles/HSPlasma.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Of course, `gcc14` supports `filesystem`, and the build runs with C++17 standard.
The issue was discovered here: https://github.com/macports/macports-ports/pull/31265#issuecomment-3931215988

--- a/.github/workflows/ci-build.yml
+++ b/.github/workflows/ci-build.yml
@@ -44,9 +44,9 @@ jobs:
     strategy:
       matrix:
         cfg:
-          - { os: macOS-13, xcode: Xcode_14.3.1 }
           - { os: macOS-14, xcode: Xcode_15.3 }
           - { os: macOS-15, xcode: Xcode_16.2 }
+          - { os: macos-26, xcode: Xcode_26.2 }
     steps:
       - uses: actions/checkout@v4
         with:

--- a/include/st_config.h.in
+++ b/include/st_config.h.in
@@ -44,10 +44,21 @@
 
 #define ST_ENUM_CONSTANT(type, name) constexpr type name = type::name
 
+#ifndef __has_include
+#   define __has_include(x) 0
+#endif
 #ifndef __has_feature
 #   define __has_feature(x) 0
 #endif
-#if __has_feature(attribute_availability_with_version_underscores) || (__has_feature(attribute_availability_with_message) && __clang__ && __clang_major__ >= 7)
+
+#if __has_include(<version>)
+#   include <version>
+#elif __has_include(<ciso646>)
+    // Before C++20, use the (otherwise empty) ciso646 header to detect the C++ runtime
+#   include <ciso646>
+#endif
+
+#if defined(_LIBCPP_VERSION) && (__has_feature(attribute_availability_with_version_underscores) || (__has_feature(attribute_availability_with_message) && __clang__ && __clang_major__ >= 7))
 #   define ST_AVAILABILITY(...)  __attribute__((availability( __VA_ARGS__ )))
 #else
 #   define ST_AVAILABILITY(...)

